x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/edit_output_flyout/use_output_form.tsx

```diff
@@ -623,6 +664,22 @@ export function useOutputForm(onSucess: () => void, output?: Output) {
               (val) => val !== ''
             ).length;
 
+            const maybeSecrets = (() => {
+              const secrets: KafkaOutput['secrets'] = {};
+
+              if (!kafkaSslKeyInput.value && kafkaSslKeySecretInput.value) {
+                secrets.ssl = {
+                  key: kafkaSslKeySecretInput.value,
+                };
+              }
+
+              if (!kafkaAuthPasswordInput.value && kafkaAuthPasswordSecretInput.value) {
+                secrets.password = kafkaAuthPasswordSecretInput.value;
+              }
+
+              return Object.keys(secrets).length ? secrets : undefined;
+            })();
+
```

Comment:
Doing this as an IIFE seems a little odd, would it make more sense to spin this off into a separate function called `hasSecrets` that accepts some parameters instead?

Reply:
I have no idea why I did it this way, I think I was being lazy because it accesses a few input variables that are in the current scope. Rectified in https://github.com/elastic/kibana/pull/169429/commits/f54dc6ba7adbf7719bd879f9d76eda7e2045a273